client/my-sites/exporter/export-card/index.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import flowRight from 'lodash/flowRight';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SpinnerButton from './spinner-button';
+import FoldableCard from 'components/foldable-card';
+import analytics from 'lib/analytics';
+import Interval, { EVERY_SECOND } from 'lib/interval';
+import AdvancedSettings from './advanced-settings';
+import {
+	advancedSettingsFetch,
+	exportStatusFetch,
+	setPostType,
+	startExport,
+} from 'state/site-settings/exporter/actions';
+import {
+	shouldShowProgress,
+	getSelectedPostType,
+	isExporting,
+} from 'state/site-settings/exporter/selectors';
+
+class ExportCard extends Component {
+	componentWillMount() {
+		this.props.advancedSettingsFetch( this.props.siteId );
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.siteId !== this.props.siteId ) {
+			this.props.advancedSettingsFetch( newProps.siteId );
+		}
+	}
+
+	trackExportClick = ( scope = 'all' ) => {
+		analytics.tracks.recordEvent( 'calypso_export_start_button_click', { scope } );
+	}
+
+	exportAll = () => {
+		this.trackExportClick();
+		this.props.startExport();
+	}
```

Comment:
Just curious, as this looks funny here inside of the `class` syntax, but why are we using fat-arrow functions and not the older function syntax?